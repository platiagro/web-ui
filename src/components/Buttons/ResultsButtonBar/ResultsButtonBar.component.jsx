import React from 'react';
import { Button } from 'antd';
import PropTypes from 'prop-types';
import { TableOutlined } from '@ant-design/icons';

const ResultsButtonBar = ({
  loading,
  disabled,
  showingResults,
  handleEditClick,
  handleResultsClick,
}) => {
  const getDisabledTitle = () => {
    return disabled
      ? 'Para visualizar os resultados, primeiro execute o treinamento.'
      : '';
  };

  return showingResults ? (
    <Button onClick={handleEditClick} shape='round' type='primary'>
      Visualizar par√¢metros
    </Button>
  ) : (
    <Button
      onClick={handleResultsClick}
      shape='round'
      type='primary-inverse'
      disabled={disabled}
      loading={loading}
      icon={<TableOutlined />}
      title={getDisabledTitle()}
      style={{ marginBottom: '14px', marginRight: '8px' }}
    >
      Visualizar resultados
    </Button>
  );
};

ResultsButtonBar.propTypes = {
  loading: PropTypes.bool,
  disabled: PropTypes.bool.isRequired,
  showingResults: PropTypes.bool.isRequired,
  handleEditClick: PropTypes.func,
  handleResultsClick: PropTypes.func.isRequired,
};

ResultsButtonBar.defaultProps = {
  loading: false,
};

export default ResultsButtonBar;
