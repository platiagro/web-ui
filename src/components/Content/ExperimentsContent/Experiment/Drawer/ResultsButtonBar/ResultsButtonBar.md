**Default**

```js
const [loading, setLoading] = React.useState(false);

const onClick = () => {
  console.log("clicked");
  setLoading(true);
  setTimeout(() => {
    setLoading(false);
  }, 2500);
};

<ResultsButtonBar
  disabled={false}
  loading={loading}
  showingResults={false}
  handleEditClick={onClick}
  handleResultsClick={onClick}
/>;
```
