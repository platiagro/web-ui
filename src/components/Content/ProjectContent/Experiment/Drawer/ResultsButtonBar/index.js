// CORE LIBS
import React from 'react';
import PropTypes from 'prop-types';

// UI LIBS
import { Button } from 'antd';

/**
 * Results Button Bar.
 * This component is responsible for displaying drawer results button bar.
 *
 * @param props
 */
const ResultsButtonBar = (props) => {
  // destructuring props
  const {
    handleEditClick,
    handleResultsClick,
    showingResults,
    disabled,
  } = props;

  // rendering component
  return showingResults ? (
    /* rendering edit or results button */
    // edit button
    <Button onClick={handleEditClick} className='ant-btn-oval' type='primary'>
      Visualizar par√¢metros
    </Button>
  ) : (
    <>
      {/* results button */}
      <Button
        onClick={handleResultsClick}
        className='ant-btn-oval'
        type='primary'
        disabled={disabled}
        title={
          disabled
            ? 'Para visualizar os resultados, primeiro execute o treinamento.'
            : ''
        }
      >
        Visualizar resultados
      </Button>
    </>
  );
};

// PROP TYPES
ResultsButtonBar.propTypes = {
  /** results button bar edit button click handler */
  handleEditClick: PropTypes.func.isRequired,
  /** results button bar results button click handler */
  handleResultsClick: PropTypes.func.isRequired,
  /** showing results drawer */
  showingResults: PropTypes.bool.isRequired,
  /** results button is disabled */
  disabled: PropTypes.bool.isRequired,
};

// EXPORT
export default ResultsButtonBar;
