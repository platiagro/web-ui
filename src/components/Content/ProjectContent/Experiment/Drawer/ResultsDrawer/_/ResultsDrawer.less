.resultsDrawer {
  .ant-tabs-tab {
    padding-left: 20px;
    padding-right: 20px;
    margin-left: 0;
    margin-right: 0;
  }
}
