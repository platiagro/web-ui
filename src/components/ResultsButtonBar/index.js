import React from 'react';

import { Button } from 'antd';

const ResultsButtonBar = ({ showResults, setShowResults }) => {
  return (
    <div>
      {showResults ? (
        <Button onClick={() => setShowResults(false)}>Editar</Button>
      ) : (
        <Button onClick={() => setShowResults(true)}>Resultados</Button>
      )}
    </div>
  );
};

export default ResultsButtonBar;
