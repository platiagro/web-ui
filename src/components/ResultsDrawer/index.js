import React, { useEffect, useState } from 'react';

import { Tag, Icon, Divider, Table, Spin } from 'antd';

import {
  getResultTable,
  getDatasetTable,
  getPlot,
} from '../../services/resultsApi';

const timeGroup = {
  none: 'Nenhum',
  daily: 'Diário',
  monthly: 'Mensal',
  dailyMonthly: 'Diário e Mensal',
};

const convertToLocaleBr = (number) => number.toLocaleString('pt-BR');

const convertMinutesToTime = (minutesReceived) => {
  const dateObj = new Date(minutesReceived * 60 * 1000);
  const hours = dateObj.getUTCHours();
  const minutes = dateObj.getUTCMinutes();
  const seconds = dateObj.getSeconds();

  const timeString = `${hours
    .toString()
    .padStart(2, '0')}:${minutes
    .toString()
    .padStart(2, '0')}:${seconds.toString().padStart(2, '0')}`;

  return timeString;
};

const fetchResultTable = async (
  experimentId,
  task,
  headerId,
  setIsLoading,
  setResultTable
) => {
  const result = await getResultTable(experimentId, task, headerId);
  setResultTable(result);
  setIsLoading(false);
};

const fetchDatasetTable = async (
  experimentId,
  datasetId,
  setIsLoading,
  setResultTable
) => {
  const result = await getDatasetTable(experimentId, datasetId);
  setResultTable(result);
  setIsLoading(false);
};

const fetchPlot = async (experimentId, setIsLoading, setPlot) => {
  const result = await getPlot(experimentId);
  setPlot(result);
  setIsLoading(false);
};

const ResultsDrawer = ({
  target,
  timeAttributes,
  attributesPreSelection,
  preType,
  genericAttributes,
  attributesFilter,
  autoML,
  table,
  tableStatistics,
  plot,
  details,
  hideDivider,
}) => {
  const [resultTable, setResultTable] = useState(null);
  const [plotDetails, setPlot] = useState(null);
  const [isLoading, setIsLoading] = useState(true);
  useEffect(() => {
    if (table && target) {
      fetchDatasetTable(
        details.uuid,
        details.datasetId,
        setIsLoading,
        setResultTable
      );
    } else if (table) {
      let task;

      if (timeAttributes) {
        task = 'feature-temporal';
      } else if (attributesPreSelection) {
        if (preType === 1) {
          task = 'pre-selection-1';
        } else {
          task = 'pre-selection-2';
        }
      } else if (genericAttributes) {
        task = 'feature-tools';
      }

      fetchResultTable(
        details.uuid,
        task,
        details.headerId,
        setIsLoading,
        setResultTable
      );
    } else if (plot) {
      fetchPlot(details.uuid, setIsLoading, setPlot);
    } else {
      setIsLoading(false);
    }
  }, []);

  return isLoading ? (
    <Spin />
  ) : (
    <div>
      {timeAttributes ? (
        <div>
          <p>Atributos agrupados: </p>
          {timeAttributes.group.map((attribute) => (
            <Tag key={attribute}>{attribute}</Tag>
          ))}

          <br />
          <br />

          <p>Período de agrupamento: </p>
          <Tag>{timeGroup[timeAttributes.period]}</Tag>
        </div>
      ) : null}

      {target ? (
        <div>
          <p>Alvo: </p>
          <Tag>{target}</Tag>
        </div>
      ) : null}

      {attributesPreSelection ? (
        <div>
          <p>Limite de dados faltantes para um atributo: </p>
          <Tag>{convertToLocaleBr(attributesPreSelection.cutoff)}</Tag>

          <br />
          <br />

          <p>Indicador máximo de correlação: </p>
          <Tag>{convertToLocaleBr(attributesPreSelection.correlation)}</Tag>
        </div>
      ) : null}

      {genericAttributes ? (
        <div>
          <p>Atributos agrupados: </p>
          {genericAttributes.group.map((attribute) => (
            <Tag key={attribute}>{attribute}</Tag>
          ))}
        </div>
      ) : null}

      {attributesFilter ? (
        <div>
          <p>Atributos excluídos: </p>
          {attributesFilter.map((attribute) => (
            <Tag key={attribute}>{attribute}</Tag>
          ))}
        </div>
      ) : null}

      {autoML ? (
        <div>
          <p>Nome do modelo: </p>
          <Tag>{details.name}</Tag>

          <br />
          <br />

          <p>Tempo de treinamento: </p>
          <Tag>{convertMinutesToTime(autoML.time)}</Tag>
        </div>
      ) : null}

      {table && resultTable ? (
        <div>
          <Divider />

          {tableStatistics ? (
            <div>
              <p>
                <strong>Atributos resultantes</strong>
              </p>
              <span>{resultTable.totalColumnsBefore}</span>
              <Icon type='arrow-right' />
              <span>{resultTable.totalColumnsAfter}</span>
              <Icon type='arrow-up' />
              <span>
                {`Mais ${resultTable.diff} atributos (+ ${resultTable.percentageDiff}%)`}
              </span>

              <br />
              <br />
            </div>
          ) : null}

          <p>
            <small>
              Exibindo 10 de&nbsp;
              {resultTable.totalLines}
              &nbsp;observações
            </small>
          </p>
          <Table
            dataSource={resultTable.rows} // {dataResult}
            columns={resultTable.header} // {columnsResult}
            size='middle'
            pagination={false}
            // scroll={{ y: 340 }}
          />
        </div>
      ) : null}

      {plot ? (
        <div>
          {hideDivider ? null : <Divider />}

          <p>
            <strong>
              {plotDetails.type === 'error-distribution'
                ? 'Gráfico de distribuição do erro'
                : 'Matriz de confusão'}
            </strong>
          </p>
          {plotDetails.type === 'error-distribution' ? (
            <p>
              O gráfico de distribuição do erro indica o desempenho de modelos
              para problemas de regressão, a partir das inferências. Para isso,
              são utilizados dados de teste com valores reais conhecidos.
            </p>
          ) : (
            <p>
              A matriz de confusão mede o desemplenho de modelos de
              classificação, a partir da frequência de previsões corretas e
              incorretas. Para isso, são utilizados dados de teste com valores
              reais conhecidos.
            </p>
          )}

          <br />
          <br />

          <img alt='plot' src={plotDetails.imageUrl} />
        </div>
      ) : null}
    </div>
  );
};

export default ResultsDrawer;
