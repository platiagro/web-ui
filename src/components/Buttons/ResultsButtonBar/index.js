import ResultsButtonBar from "./ResultsButtonBar.component";

export default ResultsButtonBar;
