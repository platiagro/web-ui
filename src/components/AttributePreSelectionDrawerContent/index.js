import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { InputNumber } from 'antd';

import InfoHelper from '../InfoHelper';
import ResultsDrawer from '../ResultsDrawer';
import ResultsButtonBar from '../ResultsButtonBar';

const infoHelperContent = (
  <div>
    <p>
      Exemplo: 0,8 indica que, se existir dois atributos com índice de
      correlação entre eles maior ou igual a esse valor, um deles será excluído
      do conjunto de dados para não prejudicar o processo de treinamento de
      modelos.
    </p>
    <p>
      Os valores abaixo indicam variáveis que podem ter:
      <br />
      entre 0,9 e 1,0: correlação muito forte;
      <br />
      entre 0,7 e 0,9: correlação forte;
      <br />
      entre 0,5 e 0,7: correlação moderada;
      <br />
      entre 0,3 e 0,5: correlação fraca;
      <br />
      entre 0,0 e 0,3: correlação muito fraca ou desprezível.
    </p>
  </div>
);

const AttributePreSelectionDrawerContent = ({
  // dataSets,
  parameter,
  setCutoff,
  setCorrelation,
  runStatus,
  taskStatus,
  details,
  preType,
}) => {
  // resultados
  const [results, setResults] = useState(
    (runStatus === 'Failed' || runStatus === 'Succeeded') &&
    taskStatus === 'Succeeded'
  );
  const [showResults, setShowResults] = useState(results);
  return (
    <div>
      {!showResults ? (
        <div>
          <p>Limite de dados faltantes para um atributo</p>
          <p>
            <small>
              Valor que indica quando um atributo será excluído do conjunto de
              dados, a partir da proporção de dados faltantes. Os valores devem
              estar entre 0,0 e 1,0
            </small>
          </p>
          <InputNumber
            value={parameter.cutoff}
            onChange={setCutoff}
            decimalSeparator=','
            min={0}
            max={1}
            step={0.1}
          />
          <InfoHelper
            width={200}
            content={
              'Exemplo: 0,6 indica que os atributos com 60% ou mais' +
              ' dos dados faltantes serão excluídos do conjunto de dados.'
            }
          />

          <br />
          <br />
          <br />

          <p>Indicador máximo de correlação</p>
          <p>
            <small>
              O indicador de correlação mostra o quanto dois atributos são
              relacionados entre si. O&nbsp;
              <span style={{ textDecoration: 'underline' }}>
                indicador máximo de correlação
              </span>
              &nbsp;define um valor como ponto de corte para exclusão dos
              atributos, baseado na correlação entre eles. Os valores devem
              estar entre 0,0 e 1,0.
            </small>
          </p>
          <InputNumber
            value={parameter.correlation}
            onChange={setCorrelation}
            decimalSeparator=','
            min={0}
            max={1}
            step={0.1}
          />
          <InfoHelper width={300} content={infoHelperContent} />
        </div>
      ) : (
          <ResultsDrawer
            details={details}
            attributesPreSelection={parameter}
            preType={preType}
            table
            tableStatistics
          />
        )}
      {runStatus === 'Failed' && taskStatus === 'Succeeded' ? (
        <ResultsButtonBar
          setShowResults={setShowResults}
          showResults={showResults}
        />
      ) : null}
    </div>
  );
};

AttributePreSelectionDrawerContent.propTypes = {
  // dataSets: PropTypes.arrayOf(PropTypes.any).isRequired,
  parameter: PropTypes.shape({
    cutoff: PropTypes.number,
    correlation: PropTypes.number,
  }).isRequired,
  setCorrelation: PropTypes.func.isRequired,
  setCutoff: PropTypes.func.isRequired,
};

export default AttributePreSelectionDrawerContent;
