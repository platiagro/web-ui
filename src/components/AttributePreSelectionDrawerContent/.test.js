import React from 'react';

import { shallow, mount } from 'enzyme'

import { act } from 'react-dom/test-utils';

import AttributePreSelectionDrawerContent from '.';
import ResultsDrawer from '../ResultsDrawer';
import ResultsButtonBar from '../ResultsButtonBar';

const parameter = {
  cutoff: 1,
  correlation: 1
}

const setCutoffPre2 = (e) => { };
const setCorrelationPre2 = (e) => { };

describe('AttributePreSelectionDrawerContent component', () => {

  it('is expected render without crashing', () => {
    shallow(<AttributePreSelectionDrawerContent
      parameter={parameter}
      setCutoff={setCutoffPre2}
      setCorrelation={setCorrelationPre2}
    />);
  });

  it('is expected render without crashing with runStatus = Failed and taskStatus = Succeeded', () => {
    act(() => {
      shallow(<AttributePreSelectionDrawerContent
        parameter={parameter}
        setCutoff={setCutoffPre2}
        setCorrelation={setCorrelationPre2}
        runStatus={'Failed'}
        taskStatus={'Succeeded'}
      />);
    });
  });

  it('is expected to be of type Div', () => {
    const wrapper = shallow(<AttributePreSelectionDrawerContent
      parameter={parameter}
      setCutoff={setCutoffPre2}
      setCorrelation={setCorrelationPre2}
    />);

    expect(wrapper.is('div')).toBeTruthy();
  });

  it('Div child is expected to have a ResultsDrawer child and ResultsButtonBar button', () => {

    act(() => {
      const wrapper = mount(<AttributePreSelectionDrawerContent
        parameter={parameter}
        setCutoff={setCutoffPre2}
        setCorrelation={setCorrelationPre2}
        details={[{ "uuid": "uuid", "headerId": "headerId" }]}
        runStatus={'Failed'}
        taskStatus={'Succeeded'}
      />);

      expect(
        wrapper
          .children('div')
          .children(ResultsDrawer)
          .exists()
      ).toBeTruthy();

      expect(
        wrapper
          .children('div')
          .children(ResultsButtonBar)
          .exists()
      ).toBeTruthy();
    });
  });

  it('is expected render html correctly', () => {
    const wrapper = shallow(<AttributePreSelectionDrawerContent
      parameter={parameter}
      setCutoff={setCutoffPre2}
      setCorrelation={setCorrelationPre2}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
