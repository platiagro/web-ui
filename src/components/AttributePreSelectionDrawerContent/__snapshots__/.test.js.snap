// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributePreSelectionDrawerContent component is expected render html correctly 1`] = `
<div>
  <div>
    <p>
      Limite de dados faltantes para um atributo
    </p>
    <p>
      <small>
        Valor que indica quando um atributo será excluído do conjunto de dados, a partir da proporção de dados faltantes. Os valores devem estar entre 0,0 e 1,0
      </small>
    </p>
    <InputNumber
      decimalSeparator=","
      max={1}
      min={0}
      onChange={[Function]}
      step={0.1}
      value={1}
    />
    <InfoHelper
      content="Exemplo: 0,6 indica que os atributos com 60% ou mais dos dados faltantes serão excluídos do conjunto de dados."
      width={200}
    />
    <br />
    <br />
    <br />
    <p>
      Indicador máximo de correlação
    </p>
    <p>
      <small>
        O indicador de correlação mostra o quanto dois atributos são relacionados entre si. O 
        <span
          style={
            Object {
              "textDecoration": "underline",
            }
          }
        >
          indicador máximo de correlação
        </span>
         define um valor como ponto de corte para exclusão dos atributos, baseado na correlação entre eles. Os valores devem estar entre 0,0 e 1,0.
      </small>
    </p>
    <InputNumber
      decimalSeparator=","
      max={1}
      min={0}
      onChange={[Function]}
      step={0.1}
      value={1}
    />
    <InfoHelper
      content={
        <div>
          <p>
            Exemplo: 0,8 indica que, se existir dois atributos com índice de correlação entre eles maior ou igual a esse valor, um deles será excluído do conjunto de dados para não prejudicar o processo de treinamento de modelos.
          </p>
          <p>
            Os valores abaixo indicam variáveis que podem ter:
            <br />
            entre 0,9 e 1,0: correlação muito forte;
            <br />
            entre 0,7 e 0,9: correlação forte;
            <br />
            entre 0,5 e 0,7: correlação moderada;
            <br />
            entre 0,3 e 0,5: correlação fraca;
            <br />
            entre 0,0 e 0,3: correlação muito fraca ou desprezível.
          </p>
        </div>
      }
      width={300}
    />
  </div>
</div>
`;
